scylla/tests/integration/compression.rs

```diff
@@ -0,0 +1,134 @@
+
+        let request_rule = |tx| {
+            RequestRule(
+                    Condition::or(Condition::RequestOpcode(RequestOpcode::Query),
+                        Condition::RequestOpcode(RequestOpcode::Execute)).and(
+                    Condition::not(Condition::ConnectionRegisteredAnyEvent)),
+                RequestReaction::noop().with_feedback_when_performed(tx),
+            )
+        };
+
+        let (request_tx, mut request_rx) = mpsc::unbounded_channel();
+        for running_node in running_proxy.running_nodes.iter_mut() {
+            running_node.change_request_rules(Some(vec![request_rule(request_tx.clone())]));
+        }
+
+        let session: Session = SessionBuilder::new()
+            .known_node(proxy_uris[0].as_str())
+            .address_translator(Arc::new(translation_map))
+            .compression(compression)
+            .build()
+            .await
+            .unwrap();
+
+        let ks = unique_keyspace_name();
+        session.ddl(format!("CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = {{'class' : 'NetworkTopologyStrategy', 'replication_factor' : 3}}", ks)).await.unwrap();
+        session.use_keyspace(ks, false).await.unwrap();
+        session
+            .ddl("CREATE TABLE test (k text PRIMARY KEY, t text, i int, f float)")
+            .await
+            .unwrap();
+
+        let q = Query::from("INSERT INTO test (k, t, i, f) VALUES (?, ?, ?, ?)");
+        let large_string = "a".repeat(text_size);
+        session.query_unpaged(q.clone(), ("key", large_string.as_str(), 42_i32, 24.03_f32)).await.unwrap();
+
+        let result: Vec<(String, String, i32, f32)> = session
+            .query_unpaged("SELECT k, t, i, f FROM test WHERE k = 'key'", &[])
+            .await
+            .unwrap()
+            .into_rows_result()
+            .unwrap()
+            .rows::<(String, String, i32, f32)>()
+            .unwrap()
+            .collect::<Result<_, _>>()
+            .unwrap();
+
+        assert_eq!(result, vec![(String::from("key"), large_string, 42_i32, 24.03_f32)]);
+
+
+        let mut total_frame_size = 0;
+        while let Ok((request_frame, _shard)) = request_rx.try_recv() {
+            total_frame_size += request_frame.body.len();
+        }
+        println!("Total frame size: {}", total_frame_size);
+        assert!(expected_frame_total_size_range.contains(&total_frame_size));
+
+        running_proxy
+
+    }).await;
+
+    match res {
+        Ok(()) => (),
+        Err(ProxyError::Worker(WorkerError::DriverDisconnected(_))) => (),
+        Err(err) => panic!("{}", err),
+    }
+}
+
+#[tokio::test]
+#[cfg(not(scylla_cloud_tests))]
+async fn should_execute_queries_without_compression() {
+    test_compression(None, 1024, 200..9000).await;
+}
+
+#[tokio::test]
+#[cfg(not(scylla_cloud_tests))]
+async fn should_execute_queries_without_compression_10mb() {
+    test_compression(None, 1024 * 10, 10000..19000).await;
+}
```

Comment:
Just to be sure: length is given in bytes, so `1024 * 10` is 10 kiB, not 10 MiB, correct? Because the name of the test is `...mb`, and this confuses me.

Reply:
yes I was confusing both thing
it's doesn't make a big difference in this tests if it's MB or MiB 

I hope it's a bit clearer now